### PR TITLE
[#5299] Allow filtering by empty string

### PR DESCRIPTION
**Pull request by Tonkonozhenko**
###### Before submitting the PR, please make sure you do the following 
-  [ ] If there are changes to the backend codebase, run the backend tests with `lein test && lein eastwood && lein bikeshed && lein docstring-checker && ./bin/reflection-linter`
-  [ ] Run the frontend and integration tests with  `yarn lint && yarn flow && yarn test`)
-  [ ] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

### CODE REVIEW

**Comment by camsaul** [comment]
Hey @Tonkonozhenko, did you run the tests for this?

File: src/metabase/util/schema.clj [hunk 1]
```diff
@@ -106,6 +106,10 @@
 ;;; |                                                 USEFUL SCHEMAS                                                 |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(def AllowedBlankString
+  "Schema for a string."
+  (s/named (s/cond-pre s/Str) (tru "String")))
```
Comment (camsaul) [inline src/metabase/util/schema.clj]: You definitely don't need to, and shouldn't define this. If you want a string that can be blank, you can just use `s/Str` directly. 

At any rate, [`cond-pre`](https://github.com/plumatic/schema#other-schema-types) is only for creating a conditional schemas from one or more schemas -- i.e. to allow something that satisfies any of several different schemas (e.g. a `String` OR an `Integer`).

**Comment by camsaul** [comment]
I'm also not 100% sure if this still applies after the Query Processor refactor from a few months ago

@Tonkonozhenko, if you'd like to rebase this against master and try running the tests, I'd be happy to take another look

**Comment by mazameli** [comment]
@Tonkonozhenko Going to close this for now since there hasn't been activity since last year. If you'd like to continue with this PR and address @camsaul's change requests and sign the CLA, we'd be happy to review this again. Thanks.

**Pull request closed without merging**